dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4878,8 +4878,34 @@ public Node visitWhileLoop(WhileLoopTree tree, Void p) {
 
             // Condition
             addLabelForNextNode(conditionStart);
+
+            boolean isWhileTrue = false;
+            int firstConditionPos = nodeList.size();
+            String output = "First condition position" + String.valueOf(firstConditionPos) + "\n";
             if (tree.getCondition() != null) {
                 unbox(scan(tree.getCondition(), p));
+                output = output + "current node counts" + String.valueOf(nodeList.size()) + "\n";
+                Node node = nodeList.get(firstConditionPos).getNode();
+                // Node node = nodeList.get(nodeList.size()-1).getNode();
+                if (node instanceof BooleanLiteralNode) {
+                    output += "contain boolean literal\n";
+                    BooleanLiteralNode booleanLiteralNode = (BooleanLiteralNode) node;
+                    if (booleanLiteralNode.getValue() == true) {
+                        output += "boolean literal is true\n";
+                        isWhileTrue = true;
+                    } else {
+                        if (nodeList.size() > firstConditionPos + 1) {
```

Comment:
Does javac treat `while (!false)` as a special case?
Is it really worth handling such unusual code?